test/libyul/yulOptimizerTests/splitJoin/control_flow.yul

```diff
@@ -1,6 +1,7 @@
 {
     if mul(add(calldataload(0), 2), 3) {
-        for { let a := 2 } lt(a, mload(a)) { a := add(a, mul(a, 2)) } {
```

Comment:
I think this change can/should be reverted if expressionSplitter does not require ForLoopInitRewriter anymore.